src/v/datalake/record_translator.cc

```diff
@@ -78,7 +79,7 @@ record_translator::build_type(std::optional<resolved_type> val_type) {
                   ret_type.fields[0]->type);
                 // Use the next id of the system defaults.
                 system_fields.fields.emplace_back(iceberg::nested_field::create(
-                  6,
+                  11,
```

Comment:
hardcoded ids seem pretty error prone, guess we need to define these constants somewhere in a future refactor.

Reply:
Agreed, couldn't think of a good refactor this time around, but this will all need to change soon anyway for the topic config switch